mmr/runner.rs

```diff
@@ -72,37 +59,38 @@ impl Runner {
         }
     }
 
+    /// Get the cache count
+    // fn cache_count(&self) -> Result<i64, Error> {
+    //     let store = Store::new(&self.path);
+    //     let res = eth_header_with_proof_caches
+    //         .filter(root.is_not_null())
+    //         .select(count(root))
+    //         .first::<i64>(&store.conn);
+    //     if let Err(e) = res {
+    //         Err(Error::Diesel(e))
+    //     } else {
+    //         Ok(res?)
+    //     }
+    // }
+
     /// Push new header hash into storage
     fn push(&mut self, pnumber: i64) -> Result<(), Error> {
         let store = Store::new(&self.path);
-        let count = self.mmr_count().unwrap_or(0);
-
         // Get Hash
         let conn = store.conn();
         let cache = eth_header_with_proof_caches
             .filter(number.eq(pnumber))
             .first::<Cache>(&store.conn)?;
 
-        let mut mmr = MMR::<_, MergeHash, _>::new(count as u64, store);
-        let rpos = mmr.push(H256::from(&cache.hash[2..]))?;
-
-        // gen mmr proof
-        let proof = mmr
-            .gen_proof(vec![rpos])?
-            .proof_items()
-            .iter()
-            .map(|h| h.hex())
-            .collect::<Vec<String>>()
-            .join(",");
+        let mut mmr = MMR::<_, MergeHash, _>::new(self.mmr_count().unwrap_or(0) as u64, store);
```

Comment:
The result of `self.mmr_count()` is equal to `last_mmr_size` while `last_leaf_index > 0`, select from database directly will be more efficient.